orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -92,6 +92,10 @@ def __init__(self, module):
         # flag to enable symbolic shape inference for dynamic shape inputs to improve performance
         self._run_symbolic_shape_infer = True
 
+        # Use torch.onnx.OperatorExportTypes.ONNX_FALLTHROUGH to allow custom autograd.Functions'.
+        # Use torch.onnx.OperatorExportTypes.ONNX if pure ONNX is needed.
+        self._onnx_export_type = kwargs['onnx_export_type']
+
```

Comment:
Do we expect to keep ONNX as the default here, or ONNX_FALLTHROUGH -- what are the downsides of using ONNX_FALLTHROUGH?  If we do require explicit opt-in, then can we make sure that the error message on calling into a [non-exported] custom autograd function is clear about the reason?

Think there's also a question for @spandantiwari @thiagocrepaldi and the front-end team : should we expose control over ORTModule features by "plumbing through" ideas like ONNX_FALLTHROUGH, or should we expose control via ORTModule-specific options and leave the exporter an implementation detail, e.g:

mod = ORTModule(mod, enable_custom_autograd_functions=True)

Reply:
Personally I would prefer enable_custom_autograd_functions.  But this should not be a blocker for the initial merge. 

@wschin  could you please help answer the first question?

Reply:
Enabling ONNX_FALLTHROUGH could accidently enable features we don't support so it's safter to keep it `ONNX`. For advanced users, we have a flag `enable_custom_autograd_function` for them to turn on:
```
def enable_custom_autograd_function(module):
    for mode in [True, False]:
        module._execution_manager(mode)._enable_custom_autograd_function = True

```